sklearn/semi_supervised/_self_training.py

```diff
@@ -143,7 +143,10 @@ class SelfTrainingClassifier(
     >>> self_training_model = SelfTrainingClassifier(svc)
     >>> self_training_model.fit(iris.data, iris.target)
     SelfTrainingClassifier(...)
-    """
+
+    For a more detailed example of usage, see:
+    :ref:`sphx_glr_auto_examples_semi_supervised_plot_self_training_varying_threshold.py`
```

Comment:
In this particular case, we only have 4 examples in the gallery. I therefore think that it is not necessary to add a link for this estimator.